apps/visualization/views.py

```diff
@@ -62,6 +63,28 @@ def get_queryset(self):
         return DataCountryLevelMostRecent.objects.filter(filters).distinct('subvariable')
 
 
+class SourceListAggViews(ListAPIView):
+    '''
+    Returns sources list.
+    It accepts String query parameters such as emergency,iso3 as filter.
+    It also accepts query parameters such as limit and offset to request number of objects
+    in a page
+    '''
+    default_limit = 10
+    serializer_class = SourceListAggSerializer
+    filterset_class = SourceListAggFilter
+    pagination_class = LimitOffsetPagination
+
+    def get_queryset(self):
+        if self.request.query_params.get('limit'):
+            limit = int(self.request.query_params.get('limit'))
+            if limit > settings.OPEN_API_MAX_PAGE_LIMIT:
+                raise ValidationError(
+                    {'error': f'Limit must be less or equal to {settings.OPEN_API_MAX_PAGE_LIMIT}'}
+                )
```

Comment:
This is duplicated, lets create base class and use in child classes

Reply:
@bimal125 Sir, is it okay if we make a method to check page limit and call it whenever needed ?